src/item.cpp

```diff
@@ -3983,10 +3966,10 @@ long item::ammo_required() const
     return 0;
 }
 
-bool item::ammo_consume( int qty, const tripoint& pos ) {
+long item::ammo_consume( long qty, const tripoint *pos ) {
```

Comment:
> The latter isn't always possible

Such as when? I don't see you using it in any place where a sensible `pos` can't be supplied, except in `process_tool` where it's most likely a bug.

> That's more complex than depending on nullptr meaning null

By an infinitesimally small amount. But it is also more consistent with the rest of the code, cleaner, less prone to crashes and will enforce passing around the position rather than just hoping it is supplied, avoiding nasty bugs that are very likely to happen otherwise. Such as the one in `process_tool`.